pkg/apis/noobaa/v1alpha1/backingstore_types.go

```diff
@@ -157,6 +158,9 @@ type AWSS3Spec struct {
 	// SSLDisabled allows to disable SSL and use plain http
 	// +optional
 	SSLDisabled bool `json:"sslDisabled,omitempty"`
+
+	// +optional
+	AWSSTSRoleARN *string `json:"awsSTSRoleARN,omitempty"`
```

Comment:
I'm not so sure it's good to have both `Secret` and `AWSSTSRoleARN` as optional. currently, we don't validate it in the admission controller. we rely on the static validation to reject a backingstore without a secret

Reply:
can we maybe use the same secret to hold either the ARN or the access keys?

Reply:
@dannyzaken The ARN is not confidential information so why complicate stuff (the need to create a secret in the first place)? Is there a benefit in doing so?

Reply:
The current implementation breaks the validations of the `AWSS3Spec`. do you have a different suggestion?

Reply:
If we are moving this into the secret I believe we will need a new field in the spec to describe what kind of auth lives in the secret so we could validate that the current auth is provided. 
Something like: AWSAuthKind (Credentials, STS) 
WDYT?

Reply:
After some thought, I agree that using a secret for ARN makes it a bit more complicated to use. especially if, in the long run, we expect STS to be the more common method.

Reply:
Let's just add validation in the admission that checks if both the secret and ARN are missing

Reply:
Added validation to check if secret and ARN are both missing